vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -668,15 +668,15 @@ impl Issuer {
             .await
             .expect("pool cap is sufficient");
 
-        assert_eq!(
-            mem.page_count(),
-            1,
-            "larger requests not currently supported"
+        assert!(
+            mem.page_count() <= 4096,
+            "Requests larger than 4096 pages not currently supported"
         );
-        let prp = Prp {
-            dptr: [mem.physical_address(0), INVALID_PAGE_ADDR],
-            _pages: None,
-        };
+
+        let prp = self
+            .make_prp(0, (0..mem.page_count()).map(|i| mem.physical_address(i)))
+            .await;
```

Comment:
note to self: `make_prp` will need to alloc memory. Does it panic if that fails? do we care?